libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -476,4 +476,13 @@ public Thread newThread(Runnable runnable) {
       };
     }
   }
+
+  @SuppressWarnings("WeakerAccess")
```

Comment:
i wonder why are we suppressing this lint warning?

Reply:
lint says that access can be package-private. But we need to call it from outside, so make it public and suppress warning

Reply:
now since this is a new public api, i've few questions:
- what happens if this method is called from the main thread?
- what happens if this method is called while the network request is in flight?
- what happens if this method is called from multiple threads at the same time?

Reply:
I think nothing happens if we call it from the main thread.
It won't affect if the request in flight, it will affect the next request.
It's not atomic, so logic could be broken if method is called from multiple threads, but maybe it's an edge case and there is no need to handle it?

or make the method synchronized?
looks like other teams don't need that method, so it will be used only by vision team.

Reply:
ok, let's make it at the very east `synchronized`

Reply:
@andrlee sorry, didn't catch, what does _at the very east_ mean? 
shall I just make it synchronized?